src/core/print.ml

```diff
@@ -94,83 +134,75 @@ and pp_term : term pp = fun oc t ->
     | Vari(x)     -> pp_tvar oc x
     | Type        -> out oc "TYPE"
     | Kind        -> out oc "KIND"
-    | Symb(s,h)   -> pp_symbol h oc s
+    | Symb(s)     -> pp_symbol oc s
     | Meta(m,e)   -> out oc "%a%a" pp_meta m pp_env e
     | Patt(_,n,e) -> out oc "$%s%a" n pp_env e
     | TEnv(t,e)   -> out oc "$%a%a" pp_term_env t pp_env e
     (* Product and abstraction (only them can be wrapped). *)
-    | Abst(a,t)   ->
+    | Abst(a,b)   ->
         if wrap then out oc "(";
-        let pp_arg oc (x,a) =
-          if !print_domains then out oc "(%a:%a)" pp_tvar x (pp `Func) a
-          else pp_tvar oc x
-        in
-        let (x,t) = Bindlib.unbind t in
-        out oc "λ%a" pp_arg (x,a);
-        let rec pp_absts oc t =
-          match unfold t with
-          | Abst(a,t) -> let (x,t) = Bindlib.unbind t in
-                         out oc " %a%a" pp_arg (x,a) pp_absts t
-          | _         -> out oc ", %a" (pp `Func) t
-        in
-        pp_absts oc t;
+        out oc "λ";
+        let (x,t) = Bindlib.unbind b in
+        if Bindlib.binder_occur b then out oc "%a" pp_tvar x else out oc "_";
+        if !print_domains then out oc ": %a" (pp `Func) a;
+        out oc ", %a" (pp `Func) t;
```

Comment:
So here you are not printing several abstractions at once anymore. Is that intended?

Reply:
Yes, by symmetry with products.

Reply:
Yeah, but when you have no domain you end up with `λ x, λ y, λ z, t` instead of `λ x y z, t`.

Reply:
Done in c1dd49e.